### ISSUE TITLE
Group Footer article links in a `ul`

### ISSUE THREAD

**Issue by huyenltnguyen** [freeCodeCamp/freeCodeCamp#52189]
### Describe the Issue

We are currently grouping the articles in the Trending Guides section using a `div`:

https://github.com/freeCodeCamp/freeCodeCamp/blob/73e3ffd79ae5929bb759ca6b0c1ce8d37f700978/client/src/components/Footer/index.tsx#L31-L45

But I think we should use `ul` and `li` to group the items instead. This helps communicate the relationship of the links better, and allows us to remove the `data-playwright-test-label` attribute from the code (ref: https://github.com/freeCodeCamp/freeCodeCamp/pull/52028#discussion_r1365195302).

Pseudo code:
```html
<h2 id="trending-guides">{t('footer.trending-guides')}</h2>
<ul aria-labelledby="trending-guides">
  <li><Link /></li>
  <li><Link /></li>
  // ...
<ul>
```

### Affected Page

https://www.freecodecamp.org/

### Expected behavior

- Links in the Trending Guides section are rendered as a `li` and are grouped within a `ul`. The change should be made to the following files:
  - Learn platform: https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/src/components/Footer/index.tsx
  - News platform: https://github.com/freeCodeCamp/news/blob/main/src/_includes/partials/footer.njk 
  - Forum platform: https://github.com/freeCodeCamp/forum-theme/blob/7e9d600053062943df3660334b3d0a674ec170ce/common/head_tag.html
- There is no visual regression
- All `data-playwright-test-label` attributes in `Footer/index.tsx` are removed (there are 2 of them in the file)
- Both Footer snapshot and e2e tests are updated accordingly
  - Bonus point if you can merge tests in this suite into a single test case: https://github.com/freeCodeCamp/freeCodeCamp/blob/73e3ffd79ae5929bb759ca6b0c1ce8d37f700978/e2e/footer.spec.ts#L49. This change is to improve cost efficiency as e2e tests are expensive to run.

**Comment by DhanushShettyH** [comment]
Can i contribute on this

**Comment by ShaunSHamilton** [comment]
Sounds like a plan. This is also a problem on `/news` and `forum.freecodecamp.org`

**Comment by ShaunSHamilton** [comment]
Here is the relevant code for the forum:
https://github.com/freeCodeCamp/forum-theme/blob/7e9d600053062943df3660334b3d0a674ec170ce/common/head_tag.html#L25

**Comment by Sacharith** [comment]
Can I contribute to the above issue?

**Comment by ShaunSHamilton** [comment]
An issue with the https://github.com/freeCodeCamp/freeCodeCamp/labels/help%20wanted label is open for contribution. The first comprehensive PR created will be reviewed and merged. We typically do not assign issues to anyone other than long-time contributors.

If you would like to contribute, and have not read the contributors docs, please do so here: https://contribute.freecodecamp.org/#/

If you have any issues with contributing, be sure to join us on the [contributors channel](https://discord.com/channels/692816967895220344/715074489422970962), or on the [contributors sub-forum](https://forum.freecodecamp.org/c/contributors/3)

**Comment by TreasureChinyimba** [comment]
This isnt open for contribution
